File: C#2/Homeworks/01-Arrays/04MaximalSequence/MaximalSequence.cs
using System;
/*
Write a program that finds the length of the maximal sequence of equal elements in an array of 
N integers.

Input
•On the first line you will receive the number N
•On the next N lines the numbers of the array will be given

Output
•Print the length of the maximal sequence

Constraints
•1 <= N <= 1024
•Time limit: 0.1s
•Memory limit: 16MB
*/

class MaximalSequence
{
     static void Main()
     {
         int n = int.Parse(Console.ReadLine());

         int count = 1;
         int maxCount = 1;

         int[] arr = new int[n];
         for (int i = 0; i < n; i++)
         {
             arr[i] = int.Parse(Console.ReadLine());
         }

         for (int i = 0; i < n-1; i++)
         {
             if (arr[i] == arr[i + 1])
             {
                 count++;
                 if (count > maxCount)
                 {
                     maxCount = count;
                 }
             }
             else
             {
                 count = 1;
             } 
         }
         Console.WriteLine(maxCount);
     }
}

File: JS/HW/Arrays/maximal-sequence.js
//Write a program that finds the length of the maximal sequence of equal elements 
//in an array of N integers.
// On the first line you will receive the number N
// On the next N lines the numbers of the array will be given

function maximalSequence(args){
    let n = +args[0],
        numbersArr = [],
        count = 1,
        maxCount = 1;

    for(let i = 1; i < args.length; i+= 1){
        numbersArr.push(+args[i]);
    }
    
    for(let i = 0; i < numbersArr.length; i+= 1){
        if(numbersArr[i+1] === numbersArr[i]){
            count+=1;
            if(count > maxCount){
                maxCount = count;
            }
        }else
        {
            count=1;
        }
    }
    console.log(maxCount);
}

maximalSequence(['10', '2', '1', '1', '2', '3', '3', '2', '2', '2', '1']);